lib/host.py

```diff
@@ -480,6 +480,7 @@ def main_sr(self):
         else:
             sr_uuid = self.pool.param_get('default-SR')
             assert sr_uuid, f"DEFAULT_SR='default' so there must be a default SR on the pool of host {self}"
+        assert sr_uuid != "<not in database>"
```

Comment:
you mean future-proof against other non-uuid strings?  are there any precedents that would make it more likely than switching to different uuids?

To me the UUID is an opaque string received from XAPI and that can be used back in other requests.  Forcing any other constraint on it would be IMHO misplaced.

Even `"<not in database>"` is a CLI-specific string not part of the API itself (and we should likely use XAPI instead of `xe` at some point).

Reply:
Yes, since we already have an example of something that could be returned by CLI that is not an UUID, rather than testing for a specific string which might change in the future, I figured we could detect any non-UUID string. But that's just a suggestion.